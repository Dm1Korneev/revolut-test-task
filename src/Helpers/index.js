import { CURRENCY_SYMBOLS } from 'Constants';

export const getCurrencySymbol = (currency) => CURRENCY_SYMBOLS[currency];

export const floorPlus = (x, n = 2) => {
  if (Number.isNaN(x)) return Number.NaN;
  if (Number.isNaN(n)) return x;

  const m = 10 ** n;

  return Math.floor(x * m) / m;
};

export const roundPlus = (x, n = 2) => {
  if (Number.isNaN(x)) return Number.NaN;
  if (Number.isNaN(n)) return x;

  const m = 10 ** n;

  return Math.round(x * m) / m;
};

