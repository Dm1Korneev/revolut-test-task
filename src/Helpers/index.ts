import { CURRENCY_SYMBOLS } from 'Constants';

export const getCurrencySymbol = (currency) => CURRENCY_SYMBOLS[currency];

export const floorPlus = (value, decimals = 2) => Number(`${Math.floor(`${value}e${decimals}`)}e-${decimals}`);

export const roundPlus = (value, decimals = 2) => Number(`${Math.round(`${value}e${decimals}`)}e-${decimals}`);
