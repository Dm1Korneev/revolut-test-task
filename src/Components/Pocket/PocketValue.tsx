import React from 'react';
import PropTypes from 'prop-types';

import { getCurrencySymbol } from 'Helpers';

const defaultProps = {
  value: 0,
};

const propTypes = {
  currency: PropTypes.string.isRequired,
  value: PropTypes.number,
};

function PocketValue(props) {
  const {
    currency, value,
  } = props;

  return (
    <span>
      {'You have '}
      <small>{getCurrencySymbol(currency)}</small>
      {value.toFixed(2)}
    </span>
  );
}

PocketValue.propTypes = propTypes;
PocketValue.defaultProps = defaultProps;

export default PocketValue;
