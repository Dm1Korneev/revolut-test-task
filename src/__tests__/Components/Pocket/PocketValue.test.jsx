import React from 'react';
import { render } from '@testing-library/react';
import PocketValue from 'Components/Pocket/PocketValue';

import * as Helpers from 'Helpers';

describe('PocketValue component', () => {
  const props = {
    currency: 'GBP',
    value: 0.09,
  };

  test('renders the component', () => {
    const spy = jest.spyOn(Helpers, 'getCurrencySymbol').mockImplementation(() => 'Â£');

    const { asFragment } = render(<PocketValue {...props} />);
    expect(asFragment()).toMatchSnapshot();
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
