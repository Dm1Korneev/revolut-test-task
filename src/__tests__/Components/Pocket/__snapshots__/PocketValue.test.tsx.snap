// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PocketValue component renders the component 1`] = `
<DocumentFragment>
  <span>
    You have 
    <small>
      Â£
    </small>
    0.09
  </span>
</DocumentFragment>
`;
